homeassistant/components/calendar/__init__.py

```diff
@@ -850,6 +859,32 @@ async def async_create_event(entity: CalendarEntity, call: ServiceCall) -> None:
 
 async def async_list_events_service(
     calendar: CalendarEntity, service_call: ServiceCall
+) -> ServiceResponse:
+    """List events on a calendar during a time range.
+
+    Deprecated: please use async_get_events_service.
+    """
+    _LOGGER.warning(
+        "Detected use of service 'calendar.list_events'. "
+        "This is deprecated and will stop working in Home Assistant 2024.6. "
+        "Use 'calendar.get_events' instead which supports multiple entities",
+    )
+    async_create_issue(
+        calendar.hass,
+        DOMAIN,
+        "deprecated_service_calendar_list_events",
+        breaks_in_ha_version="2024.6.0",
+        is_fixable=False,
```

Comment:
We should set this to fixable as the users can change their automations/scripts/etc.
Example: https://github.com/home-assistant/core/blob/2023.9.0/homeassistant/components/counter/__init__.py#L289-L300